src/DynamoRevit/Models/RevitNodeModel.cs

```diff
@@ -1,8 +1,14 @@
-﻿using Dynamo.Graph.Nodes;
+﻿using System.Collections.Generic;
+using Dynamo.Graph.Nodes;
+using Newtonsoft.Json;
 
 namespace Dynamo.Applications.Models
 {
     public abstract class RevitNodeModel : NodeModel
     {
+        public RevitNodeModel() { }
```

Comment:
you needed to add this constructor for what reason?

Reply:
If I only add the json constructor to  `RevitNodeModel` all non-json constructors in classes that inherit from `RevitNodeModel` point at the Json constructor in `RevitNodeModel` causing an error due to the differing parameters each constructor takes.  If I don't add either constructor (json or non-json) the children can no longer call the base constructor which in turn would call the `NodeModel` constructor.  `ElementsOfFamilyType : ElementsQueryBase : RevitNodeModel : NodeModel`

Reply:
There may be a more efficient way of doing this but seems this is how it has been done in the past which requires the interim classes to all has json constructors as well. ex:
```C#
[JsonConstructor]
        public SomeRevitUINode(IEnumerable<PortModel> inPorts, IEnumerable<PortModel> outPorts) : base(inPorts, outPorts)
        {
        }
```